tests/cases/py_client/lib_test.py

```diff
@@ -555,6 +555,26 @@ def mock(url, request):
             self.assertTrue(obj.getvalue().endswith(expected))
             self.assertEqual(len(hits), 2)
 
+    def testDownloadFail(self):
+        # Create item
+        item = self.client.createItem(self.publicFolder['_id'],
+                                      'SomethingMostUnique')
+        # Upload file to item
+        path = os.path.join(self.libTestDir, 'sub0', 'f')
+        file = self.client.uploadFileToItem(item['_id'], path)
+
+        obj = six.BytesIO()
+
+        @httmock.urlmatch(path=r'.*.*/file/.+/download$')
+        def mock(url, request):
+            resp = request.models.Response()
+            resp.status_code = 500
+            resp.text = "error"
+            return resp
+
+        # Attempt to download file to object stream, should raise HttpError
+        self.assertRaises(girder_client.HttpError, self.client.downloadFile(file['_id'], obj))
```

Comment:
This isn't actually asserting that the exception is raised since the usage of `self.assertRaises` requires the callable and `*args` and `**kwargs`. You can grep for other usages in Girder, but I would recommend using it as a context manager:
```
with self.assertRaises(girder_client.HttpError):
    self.client.downloadFile(file['_id'], obj)
```